### PR TITLE
adds ip subnet utility functions to network_common

### PR DESCRIPTION
**Pull request by privateip**
* adds functions to convert between netmask and masklen
* adds functions to verify netmask and masklen
* adds function to determine network and subnet from address / mask pair

### CODE REVIEW

**Comment by dagwieers** [comment]
This is (partly) functionality I wanted to add myself as well. However I would rely on netaddr rather than implementing it myself.

**Comment by bcoca** [comment]
i'm thinking we need to unify code with the filters also

**Comment by privateip** [comment]
Yeah agreed with all points here in.   My only sticking point would be that `netaddr` is currently not an installation dependency for Ansible.  So in order to move this forward, I would assert that `netaddr` needs to become a dependency

**Comment by bcoca** [comment]
i would add it to 'recommends' vs dependancy, its only needed for a subset, we should of course make strong note that if you are using ansible for networking ... you need it.

File: test/units/module_utils/test_network_common.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+__metaclass__ = type
+
+import os
+import json
+
+from ansible.compat.tests import unittest
+from ansible.compat.tests.mock import patch, MagicMock
+from ansible.errors import AnsibleModuleExit
+from ansible.module_utils import network_common as net
+from ansible.module_tuils.network_common import ComplexDict
```
Comment (mattclay) [inline test/units/module_utils/test_network_common.py:30]: Typo in `module_utils`.

**Comment by mattclay** [comment]
CI failure due to the aforementioned typo and the following test failure:

```
2017-02-07 23:14:46 TASK [ios_command : test contains operator] ************************************
2017-02-07 23:14:46 task path: /root/src/github.com/ansible/ansible/test/integration/targets/ios_command/tests/cli/contains.yaml:4
2017-02-07 23:14:46 <ec2-52-91-221-156.compute-1.amazonaws.com> closing connection
2017-02-07 23:14:46 An exception occurred during task execution. The full traceback is:
2017-02-07 23:14:46 Traceback (most recent call last):
2017-02-07 23:14:46   File "/root/src/github.com/ansible/ansible/lib/ansible/executor/task_executor.py", line 126, in run
2017-02-07 23:14:46     res = self._execute()
2017-02-07 23:14:46   File "/root/src/github.com/ansible/ansible/lib/ansible/executor/task_executor.py", line 504, in _execute
2017-02-07 23:14:46     result = self._handler.run(task_vars=variables)
2017-02-07 23:14:46   File "/root/src/github.com/ansible/ansible/lib/ansible/plugins/action/network.py", line 43, in run
2017-02-07 23:14:46     module.main()
2017-02-07 23:14:46   File "/root/src/github.com/ansible/ansible/lib/ansible/modules/network/ios/ios_command.py", line 228, in main
2017-02-07 23:14:46     commands = parse_commands(module, warnings)
2017-02-07 23:14:46   File "/root/src/github.com/ansible/ansible/lib/ansible/modules/network/ios/ios_command.py", line 188, in parse_commands
2017-02-07 23:14:46     response=dict()
2017-02-07 23:14:46 TypeError: __init__() takes exactly 3 arguments (2 given)
2017-02-07 23:14:46 
2017-02-07 23:14:46 fatal: [ios-csr1000v]: FAILED! => {
2017-02-07 23:14:46     "failed": true, 
2017-02-07 23:14:46     "msg": "Unexpected failure during module execution.", 
2017-02-07 23:14:46     "stdout": ""
2017-02-07 23:14:46 }
```

**Pull request closed without merging**